addons/website_sale/models/product_template.py

```diff
@@ -157,16 +157,16 @@ def _sort_key_variant(variant):
 
         return self._get_possible_variants(parent_combination).sorted(_sort_key_variant)
 
-    def _get_sales_prices(self, pricelist):
-        pricelist.ensure_one()
+    def _get_sales_prices(self, currency, pricelist):
```

Comment:
```suggestion
    def _get_sales_prices(self, pricelist=None):
```

to clearly specify it's sometimes not provided (or empty)
Also, if pricelist isn't specified, fallback to self.env.company.currency_id instead of using a param ?